frontend/src/concepts/pipelines/content/artifacts/ArtifactUriLink.tsx

```diff
@@ -1,46 +1,38 @@
 import React from 'react';
 import { Button, Flex, FlexItem, Icon, Popover, Truncate } from '@patternfly/react-core';
 import { ExclamationTriangleIcon, ExternalLinkAltIcon } from '@patternfly/react-icons';
-import { usePipelinesAPI } from '~/concepts/pipelines/context';
-import { useIsAreaAvailable, SupportedArea } from '~/concepts/areas';
-import { MAX_STORAGE_OBJECT_SIZE, fetchStorageObjectSize } from '~/services/storageService';
+import { MAX_STORAGE_OBJECT_SIZE } from '~/services/storageService';
 import { bytesAsRoundedGiB } from '~/utilities/number';
 import { ArtifactType } from '~/concepts/pipelines/kfTypes';
-import { extractS3UriComponents, getArtifactUrlFromUri } from './utils';
+import { useArtifactStorage } from '~/concepts/pipelines/apiHooks/useArtifactStorage';
+import { Artifact } from '~/third_party/mlmd';
 
 interface ArtifactUriLinkProps {
   uri: string;
   type: string;
+  artifact: Artifact;
 }
 
-export const ArtifactUriLink: React.FC<ArtifactUriLinkProps> = ({ uri, type }) => {
-  const { namespace } = usePipelinesAPI();
-  const isS3EndpointAvailable = useIsAreaAvailable(SupportedArea.S3_ENDPOINT).status;
-  const [size, setSize] = React.useState<number | null>(null);
+export const ArtifactUriLink: React.FC<ArtifactUriLinkProps> = ({ uri, type, artifact }) => {
   const isClassificationMetrics = type === ArtifactType.CLASSIFICATION_METRICS;
+  const [url, setUrl] = React.useState<string | undefined>();
+  const [size, setSize] = React.useState<number | undefined>();
 
-  const url = React.useMemo(() => {
-    if (!uri || !isS3EndpointAvailable) {
-      return;
-    }
-
-    const uriComponents = extractS3UriComponents(uri);
+  const artifactStorage = useArtifactStorage();
 
-    if (uriComponents) {
-      fetchStorageObjectSize(namespace, uriComponents.path)
-        .then((sizeBytes) => setSize(sizeBytes))
-        .catch(() => null);
+  React.useEffect(() => {
+    if (artifactStorage.enabled) {
+      artifactStorage.getStorageObjectUrl(artifact).then((u) => setUrl(u));
+      artifactStorage.getStorageObjectSize(artifact).then((s) => setSize(s));
     }
-
-    return getArtifactUrlFromUri(uri, namespace);
-  }, [isS3EndpointAvailable, namespace, uri]);
+  }, [artifact, artifactStorage]);
 
   if (!url || isClassificationMetrics) {
     return uri;
   }
 
   // we do not fetch over 100MB
-  const isOversizedFile = size !== null && size > MAX_STORAGE_OBJECT_SIZE;
+  const isOversizedFile = !!size && size > MAX_STORAGE_OBJECT_SIZE;
 
   return (
```

Comment:
we can remove the size constraint here becuase they are downloading it and we are not. so just remove